pkg/controller/realmadmin/show.go

```diff
@@ -16,36 +16,97 @@ package realmadmin
 
 import (
 	"context"
-	"encoding/csv"
+	"encoding/json"
+	"fmt"
 	"net/http"
 	"strconv"
+	"strings"
 	"time"
 
+	"github.com/google/exposure-notifications-verification-server/internal/icsv"
 	"github.com/google/exposure-notifications-verification-server/pkg/cache"
 	"github.com/google/exposure-notifications-verification-server/pkg/controller"
 	"github.com/google/exposure-notifications-verification-server/pkg/database"
 )
 
-var cacheTimeout = 5 * time.Minute
+const cacheTimeout = 30 * time.Minute
 
-// ResultType specifies which type of renderer you want.
-type ResultType int
+func (c *Controller) HandleShow() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
 
-const (
-	HTML ResultType = iota
-	JSON
-	CSV
-)
+		realm := controller.RealmFromContext(ctx)
+		if realm == nil {
+			controller.MissingRealm(w, r, c.h)
+			return
+		}
+
+		now := time.Now().UTC()
+		past := now.Add(-30 * 24 * time.Hour)
+
+		pth := r.URL.Path
+		switch {
+		case strings.HasSuffix(pth, ".csv"):
+			var filename string
+			var stats icsv.Marshaler
+			var err error
+
+			nowFormatted := now.Format("20060102150405")
+
+			switch r.URL.Query().Get("scope") {
+			case "external":
+				filename = fmt.Sprintf("%s-external-issuer-stats.csv", nowFormatted)
+				stats, err = c.getExternalIssuerStats(ctx, realm, now, past)
+			case "user":
+				filename = fmt.Sprintf("%s-user-stats.csv", nowFormatted)
+				stats, err = c.getUserStats(ctx, realm, now, past)
+			default:
+				filename = fmt.Sprintf("%s-realm-stats.csv", nowFormatted)
+				stats, err = c.getRealmStats(ctx, realm, now, past)
+			}
+
+			if err != nil {
+				controller.InternalError(w, r, c.h, err)
+				return
+			}
+
+			c.h.RenderCSV(w, http.StatusOK, filename, stats)
+		case strings.HasSuffix(pth, ".json"):
+			var stats json.Marshaler
+			var err error
+
+			switch r.URL.Query().Get("scope") {
```

Comment:
don't see nested switches much

Reply:
I might pull the JSON/CSV method out into its own function in a later PR.